### PR TITLE
chore: Storybook 8 upgrade

### PR DESCRIPTION
**Pull request by dlnr**
Also tried to resolve the vite cjs warning but ran into jest issues.

### CODE REVIEW

File: storybook/config/main.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import type { StorybookConfig } from '@storybook/react-vite'
-import remarkGfm from 'remark-gfm'
```
Comment (VincentSmedinga) [inline storybook/config/main.ts:2]: I now understand why we need this – it allows us to use GitHub Flavoured Markdown in Storybook.

For example, our tables no longer work without: https://amsterdam.github.io/design-system/demo-DES-618-storybook-8/?path=/docs/docs-design-guidelines-grid--docs#spacious-2

> This is a plugin to enable the extensions to markdown that GitHub adds with GFM: autolink literals, footnotes, strikethrough, tables, and tasklists. These extensions by GitHub to CommonMark are called [GFM](https://github.github.com/gfm/) (GitHub Flavored Markdown).

> Users frequently believe that some of these extensions, specifically autolink literals and tables, are part of normal markdown, so using remark-gfm will help match your implementation to their understanding of markdown.

– [source](https://github.com/remarkjs/remark-gfm)

Comment (dlnr) [reply storybook/config/main.ts:2]: Thanks for the headsup, it works with the latest version.

**Review by VincentSmedinga — APPROVED** [review]

File: storybook/config/manager-head.html [hunk 2]
```diff
@@ -1,20 +1,9 @@
 <script id="storybook-mod-expand-all-folders" type="text/javascript">
-  // Somewhat hacky way to expand all folders. Storybook does not support that natively.
```
Comment (alimpens) [inline storybook/config/manager-head.html:2]: This is still a hacky way to do this, right? I'd keep the comment

File: storybook/package.json [hunk 3]
```diff
@@ -26,34 +26,32 @@
...
     "react-dom": "18.2.0",
     "require-from-string": "2.0.2",
-    "storybook": "7.6.17",
+    "storybook": "8.0.0",
     "vite": "5.1.4"
   },
   "dependencies": {
-    "remark-gfm": "3.0.1"
+    "@storybook/test": "8.0.0",
+    "remark-gfm": "4.0.0"
```
Comment (alimpens) [inline storybook/package.json:55]: These should be devDependencies as well, I think?

File: storybook/package.json [hunk 4]
```diff
@@ -26,34 +26,32 @@
...
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "require-from-string": "2.0.2",
-    "storybook": "7.6.17",
+    "storybook": "8.0.0",
     "vite": "5.1.4"
   },
   "dependencies": {
-    "remark-gfm": "3.0.1"
+    "@storybook/test": "8.0.0",
```
Comment (alimpens) [inline storybook/package.json:54]: We don't seem to be using this (yet)

File: storybook/package.json (hunk 3 above)
Comment (dlnr) [reply storybook/package.json:55]: It was in the deps previously, and is used in the build. However it's probably baked in during the static build so I'll move it.

File: storybook/package.json (hunk 4 above)
Comment (dlnr) [reply storybook/package.json:54]: Sure, I've added to replace @storybook/testing-library, which was also unused I guess? After removing my interaction tests for Dialog?

Comment (alimpens) [reply storybook/package.json]: Yeah, I think so

**Review by alimpens — APPROVED** [review]

**Pull request merged**